### ISSUE TITLE
[bug] test failure with specific seed indicates cross-pollution

### ISSUE THREAD

**Issue by flavorjones** [sparklemotion/nokogiri#2164]
**Please describe the bug**

The CI test run at https://ci.nokogiri.org/teams/nokogiri-core/pipelines/nokogiri/jobs/cruby-gem-test/builds/66 failed, and a rerun passed. We don't generally have flaky tests, so I investigated.

Thanks to `minitest-bisect` this was a cinch to narrow down to:

```
Culprit methods: ["Nokogiri::XML::TestDocument#test_xpath_syntax_error"]
...
# Running tests with run options --seed=59263 -n "/^(?:Nokogiri::XML::TestDocument#(?:test_xpath_syntax_error)|Nokogiri::XML::SAX::TestParserContext#(?:test_from_file))$/":
```

**Help us reproduce what you're seeing**

You should be able to repro by running:

```
TESTOPTS='--seed=59263 "-n/^(?:Nokogiri::XML::TestDocument#(?:test_xpath_syntax_error)|Nokogiri::XML::SAX::TestParserContext#(?:test_from_file))$/"' bundle exec rake test
```

**Comment by flavorjones** [comment]
Oh, and I'm hoping to land fixes for #1610 in the v1.11/v1.12 timeframe.

**Comment by flavorjones** [comment]
The commit I mentioned above is in a PR sitting at https://github.com/sparklemotion/nokogiri/pull/2096
